core/services/relay/customendpoint/relay.go

```diff
@@ -0,0 +1,163 @@
+	PayloadType string `json:"payloadType"` // required
+
+	// Fields specific to Bridge type targets
+	BridgeRequestData string `json:"bridgeRequestData"`
+	BridgeInputAtKey  string `json:"bridgeInputAtKey"`
+
+	// The multiplier used in the job spec for storing price feed.
+	// Must be a positive integer, which is a power of 10.
+	// This should be same as the value used in the multiply task
+	// of the Observation phase while reporting final result.
+	MultiplierUsed int32 `json:"multiplierUsed"`
+}
+
+type OCR2Spec struct {
+	RelayConfig
+	ID          int32
+	IsBootstrap bool
+}
+
+// Relayer for customendpoint.
+// Note that a customendpoint integration doesn't have any associated Chain.
+// We are just uploading to some custom endpoint. This relayer is an interface to
+// doing that via OCR2. The implementation just has basic functionality needed
+// to make OCR2 work, without any associated chain.
+type Relayer struct {
+	lggr        logger.Logger
+	config      config.GeneralConfig
+	pipelineORM pipeline.ORM
+	clock       utils.Nower
+}
+
+func NewRelayer(lggr logger.Logger,
+	config config.GeneralConfig,
+	pipelineORM pipeline.ORM,
+	clock utils.Nower) *Relayer {
+	return &Relayer{
+		lggr:        lggr,
+		config:      config,
+		pipelineORM: pipelineORM,
+		clock:       clock,
+	}
+}
+
+func (r *Relayer) Start(context.Context) error {
+	return nil
+}
+
+func (r *Relayer) Close() error {
+	return nil
+}
+
+func (r *Relayer) Ready() error {
+	return nil
+}
+
+func (r *Relayer) Healthy() error {
+	return nil
+}
+
+type ocr2Provider struct {
+	configDigester offchainConfigDigester
+	reportCodec    evmreportcodec.ReportCodec
+	tracker        *contractTracker
+}
+
+// NewOCR2Provider creates a new OCR2ProviderCtx instance.
+func (r *Relayer) NewOCR2Provider(externalJobID uuid.UUID, s interface{}) (relaytypes.OCR2ProviderCtx, error) {
+	var provider ocr2Provider
+	spec, ok := s.(OCR2Spec)
+	if !ok {
+		return &provider, errors.New("unsuccessful cast to 'customendpoint.OCR2Spec'")
+	}
+	if spec.MultiplierUsed < 1 {
+		return &provider, errors.New("invalid multiplierUsed in 'customendpoint.OCR2Spec'")
+	}
+
+	digester := offchainConfigDigester{
+		EndpointName:   spec.EndpointName,
+		EndpointTarget: spec.EndpointTarget,
+		PayloadType:    spec.PayloadType,
+	}
+	codec := evmreportcodec.ReportCodec{}
+	tracker := NewTracker(spec, digester, r.lggr, r.pipelineORM, r.config, codec, r.clock)
+
+	if spec.IsBootstrap {
+		// Return early if bootstrap node (doesn't require the full OCR2 provider)
+		return &ocr2Provider{
+			configDigester: digester,
+			tracker:        &tracker,
+		}, nil
+	}
+
+	return &ocr2Provider{
+		configDigester: digester,
+		reportCodec:    codec,
+		tracker:        &tracker,
+	}, nil
+}
+
+func (p *ocr2Provider) Start(context.Context) error {
+	return p.tracker.Start()
+}
+
+func (p *ocr2Provider) Close() error {
+	return p.tracker.Close()
+}
+
+func (p ocr2Provider) Ready() error {
```

Comment:
Should probably align these method receivers:
```suggestion
func (p *ocr2Provider) Ready() error {
```

Reply:
Thanks!